packages/components/src/mobile/keyboard-aware-flat-list/test/use-scroll-to-text-input.native.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * External dependencies
+ */
+
+import { renderHook } from '@testing-library/react-native';
+
+/**
+ * Internal dependencies
+ */
+import useScrollToTextInput from '../use-scroll-to-text-input';
+
+describe( 'useScrollToTextInput', () => {
+	it( 'scrolls up to the current TextInput offset', () => {
+		// Arrange
+		const currentCaretData = { caretHeight: 10 };
+		const extraScrollHeight = 50;
+		const keyboardOffset = 100;
+		const scrollEnabled = true;
+		const scrollViewRef = { current: { scrollTo: jest.fn() } };
+		const scrollViewMeasurements = { current: { height: 600 } };
+		const scrollViewYOffset = { value: 150 };
+		const textInputOffset = 50;
+
+		const { result } = renderHook( () =>
+			useScrollToTextInput(
+				extraScrollHeight,
+				keyboardOffset,
+				scrollEnabled,
+				scrollViewMeasurements,
+				scrollViewRef,
+				scrollViewYOffset
+			)
+		);
+
+		// Act
+		result.current[ 0 ]( currentCaretData, textInputOffset );
+
+		// Assert
+		expect( scrollViewRef.current.scrollTo ).toHaveBeenCalledWith( {
+			y: textInputOffset,
+			animated: true,
+		} );
+	} );
+
+	it( 'scrolls down to the current TextInput offset', () => {
+		// Arrange
+		const currentCaretData = { caretHeight: 10 };
+		const extraScrollHeight = 50;
+		const keyboardOffset = 100;
+		const scrollEnabled = true;
+		const scrollViewRef = { current: { scrollTo: jest.fn() } };
+		const scrollViewMeasurements = { current: { height: 600 } };
+		const scrollViewYOffset = { value: 250 };
+		const textInputOffset = 750;
+
+		const { result } = renderHook( () =>
+			useScrollToTextInput(
+				extraScrollHeight,
+				keyboardOffset,
+				scrollEnabled,
+				scrollViewMeasurements,
+				scrollViewRef,
+				scrollViewYOffset
+			)
+		);
+
+		// Act
+		result.current[ 0 ]( currentCaretData, textInputOffset );
+
+		// Assert
+		const expectedYOffset =
+			textInputOffset -
+			( scrollViewMeasurements.current.height -
+				( keyboardOffset +
+					extraScrollHeight +
+					currentCaretData.caretHeight ) );
+		expect( scrollViewRef.current.scrollTo ).toHaveBeenCalledWith( {
+			y: expectedYOffset,
+			animated: true,
+		} );
+	} );
+
+	it( 'does not scroll when the ScrollView ref is not available', () => {
+		// Arrange
+		const currentCaretData = { caretHeight: 10 };
+		const extraScrollHeight = 50;
+		const keyboardOffset = 100;
+		const scrollEnabled = true;
+		const scrollViewRef = { current: null };
+		const scrollViewMeasurements = { current: { height: 600 } };
+		const scrollViewYOffset = { value: 0 };
+		const textInputOffset = 50;
+
+		const { result } = renderHook( () =>
+			useScrollToTextInput(
+				extraScrollHeight,
+				keyboardOffset,
+				scrollEnabled,
+				scrollViewMeasurements,
+				scrollViewRef,
+				scrollViewYOffset
+			)
+		);
+
+		// Act
+		result.current[ 0 ]( currentCaretData, textInputOffset );
+
+		// Assert
+		expect( scrollViewRef.current ).toBeNull();
```

Comment:
You're right, I updated the test in https://github.com/WordPress/gutenberg/pull/48791/commits/fd42475f80cf7e270d05dfba95d22e6f490f874e thanks!

Reply:
I'm checking https://github.com/WordPress/gutenberg/commit/fd42475f80cf7e270d05dfba95d22e6f490f874e and realized that if `scrollViewRef.current` is `null` the `scrollTo` shouldn't even be defined. So, probably we shouldn't mock `scrollTo`. I'm thinking about what we'd need to assert in this test but since the hook does nothing when`scrollViewRef.current` is `null`, there's no act that we can do 😅.

Reply:
😅 haha same! If `current` is null, there's no way the scrollTo method can be called. I've should've realized that when I was working on a way to mock it 😆 

I'll revert this change!